libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,116 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
```

Comment:
Probably will rewrite to have this as `DropSectionKind` and `DropSection` as a vector of `DropSectionKind`. In that case the entire code can be rewritten somewhat, potentially with a single loop, though one must be careful when removing sections with a lower index.

Reply:
After this rewrite I could also introduce `with_preset` with `ewasm` which removes the namessection and all custom sections with names `debug`.